### ISSUE TITLE
Clarify documentation about CIBW_DEPENDENCY_VERSIONS on Linux

### ISSUE THREAD

**Issue by reynoldsnlp** [pypa/cibuildwheel#1446]
### Description

TLDR; The documentation isn't clear enough about how to get the `latest` for `CIBW_DEPENDENCY_VERSIONS` in the linux builds. It might be helpful to have a table of available versions of major build tools (`build`, `flit`, `hatch`, `hatchling`, `pip`, etc.) for each container.

I can successfully build my package locally with latest versions of `build`(0.10.0), `hatch`(1.6.3), `hatchling`(1.13.0), etc., but when I try to build it using `cibuildwheel --platform linux`, I get the following error:

```
Traceback (most recent call last):
  File "/opt/python/cp36-cp36m/lib/python3.6/site-packages/pep517/in_process/_in_process.py", line 351, in <module>
    main()
  File "/opt/python/cp36-cp36m/lib/python3.6/site-packages/pep517/in_process/_in_process.py", line 333, in main
    json_out['return_val'] = hook(**hook_input['kwargs'])
  File "/opt/python/cp36-cp36m/lib/python3.6/site-packages/pep517/in_process/_in_process.py", line 250, in build_wheel
    metadata_directory)
  File "/tmp/build-env-n354s9pu/lib/python3.6/site-packages/hatchling/build.py", line 41, in build_wheel
    return os.path.basename(next(builder.build(wheel_directory, ['standard'])))
  File "/tmp/build-env-n354s9pu/lib/python3.6/site-packages/hatchling/builders/plugin/interface.py", line 76, in build
    self.metadata.core.validate_fields()
  File "/tmp/build-env-n354s9pu/lib/python3.6/site-packages/hatchling/metadata/core.py", line 99, in core
    metadata_hooks = self.hatch.metadata.hooks
  File "/tmp/build-env-n354s9pu/lib/python3.6/site-packages/hatchling/metadata/core.py", line 1245, in hooks
    raise ValueError('Unknown metadata hook: {}'.format(hook_name))
ValueError: Unknown metadata hook: vcs

ERROR Backend subprocess exited when trying to invoke build_wheel
```

I have tried using `build-system.requires` in the pyproject.toml and `pip install --upgrade` in `CIBW_BEFORE_BUILD`, but pip in the linux container can't see the latest versions at all. I don't know enough about pip/PyPI to understand how that's possible, but it appears to be the case.

The [documentation about CIBW_DEPENDENCY_VERSIONS](https://cibuildwheel.readthedocs.io/en/latest/options/#dependency-versions) says...

> This option does not affect the tools used on the Linux build - those versions are bundled with the manylinux/musllinux image that cibuildwheel uses. To change dependency versions on Linux, use the [CIBW_MANYLINUX_ / CIBW_MUSLLINUX_](https://cibuildwheel.readthedocs.io/en/latest/options/#linux-image) options.

...but I follow the link, and I don't see anything about dependency versions. If the only way to upgrade those dependencies is by using a different `manylinux`/`musllinux` container, then it would be nice to add a table to the documentation that shows which versions of the build tools are available on which containers.

### Build log

https://github.com/reynoldsnlp/pip_inside/actions/runs/4555961317/jobs/8035774205

### CI config

https://github.com/reynoldsnlp/pip_inside/actions/runs/4555961317/workflow

**Comment by henryiii** [comment]
Where is the binary build? All I see is Python files, and hatchling doesn't support binary builds without a plugin. Cibuildwheel is only for building binary wheels; pure Python wheels can be built directly with `build`, there's no issue distributing them like there is with binary wheels.

And the problem you hit above is that Python 3.6 was dropped by hatchling some time ago, and I guess it was either before vcs hooks were supported, or there's a mismatch in when `hatch-vcs` dropped 3.6 and some sort of API change.

`CIBW_DEPENDENCY_VERSIONS` is basically unused if you have a pyproject.toml. ("Basically" because it still affects the repair step and version of pip/build being used to do the build)

**Comment by henryiii** [comment]
> then it would be nice to add a table to the documentation that shows which versions of the build tools are available on which containers.

These change every release (manylinux updates weekly!), so adding a static table here would be challanging. And for users using pyproject.toml (or the upcoming version of Pip!), it really doesn't matter, everything is controlled by the pyproject.toml build-system.requires field. `CIBW_DEPENDENCY_VERSIONS` is more of an artifact of old-style builds.

**Comment by reynoldsnlp** [comment]
@henryiii how embarrassing! I've used `cibuildwheel` for other projects that have C++ backends, and mindlessly went to use it for this one, too. Thanks!
